modules/irc/src/main/ZulipClient.scala

```diff
@@ -40,15 +55,23 @@ final private class ZulipClient(ws: StandaloneWSClient, config: ZulipClient.Conf
             )
           )
           .flatMap {
-            case res if res.status == 200 => funit
-            case res                      => fufail(s"[zulip] $msg ${res.status} ${res.body}")
+            case res if res.status == 200 =>
+              (res.body[JsValue] \ "id").validate[ZulipMessage.ID] match {
+                case JsSuccess(result, _) => fuccess(result.some)
+                case JsError(err)         => fufail(s"[zulip]: $err, $msg ${res.status} ${res.body}")
+              }
+            case res => fufail(s"[zulip] $msg ${res.status} ${res.body}")
           }
           .monSuccess(_.irc.zulip.say(msg.stream))
           .logFailure(lila.log("zulip"))
           .nevermind
-        funit // return immediately! don't make mod actions slower because of that
+        if (returnMsgId) {
+          msgId
+        } else {
+          fuccess(None) // return immediately! don't make mod actions slower because of that
+        }
```

Comment:
Hopefully this structure allows to return immediately when `msgId` is not requested but I'm not sure.

Reply:
it does! Still a bit weird to have this `returnMsgId`. I would just have that function always return the future `msgId`, which can then be discarded by functions that ignore it and return immediately, i.e. `apply`